feat(formatter): support printing binary like expression

Dunqing opened the pull request:
Support printing `BinaryExpression` and `LogicalExpression`, the work is done except printing comments.

Dunqing commented:
It looks like spending a lot of time on handling IR, and IR is correct, so that performance regression is expected.

Dunqing commented on crates/oxc_formatter/src/utils/mod.rs:
```diff
@@ -0,0 +1,28 @@
+    formatter::{Formatter, prelude::soft_line_break_or_space},
+};
+
+/// This function is in charge to format the call arguments.
```
Copied from Biome, no need to change

Boshen commented:
### Merge activity

* **May 23, 7:32 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
feat(formatter): support printing binary like expression (#11185)

Support printing `BinaryExpression` and `LogicalExpression`, the work is done except printing comments.
